### PR TITLE
[FREELDR] Improvements for NEC PC-98 port

### PR DESCRIPTION
**Pull request by binarymaster**
## Purpose

Several improvements for the NEC PC-98 port. This makes it look authentic and improves early boot performance.

JIRA issue: None

## Proposed changes

- Use BIOS font and accelerated text drawing on NEC PC-98. Both features are switchable via global variables.
- Add variable for default text color. On PC/AT-compatible machines it's Gray, and on NEC PC-98 series it's White.

#### Before
image

#### Classic (minimal UI)
image

#### Eye-candy (retro)
image

More pics available [in the chat](https://chat.reactos.org/reactos/pl/weaijasmciyo8by8g1m6chrr5w).

Cc @disean :wink:

### CODE REVIEW

File: boot/freeldr/freeldr/arch/i386/pc98/pc98video.c [hunk 1]
```diff
@@ -336,4 +453,13 @@ Pc98VideoPrepareForReactOS(VOID)
     Int386(0x18, &Regs, &Regs);
 
     Pc98VideoHideShowTextCursor(FALSE);
+
+    if (UseCGFont && CGAccelDraw)
```
Comment (HBelusca) [inline boot/freeldr/freeldr/arch/i386/pc98/pc98video.c:457]:
```suggestion
    /* Clear the screen, resetting to default attributes */
    if (UseCGFont && CGAccelDraw)
```
or something like that :)

**Review by HBelusca — APPROVED** [review]
LGTM

**Review by tkreuzer — APPROVED** [review]

File: boot/freeldr/freeldr/arch/i386/pc98/pc98video.c [hunk 2]
```diff
@@ -232,6 +241,77 @@ Pc98VideoHideShowTextCursor(BOOLEAN Show)
...
+        case 13: // Light Magenta
+            return GDC_ATTR_PURPLE;
+        case 6: // Brown
+        case 14:
+            return GDC_ATTR_YELLOW;
+        case 9:
+        case 11: // Light Cyan
+            return GDC_ATTR_LIGHTBLUE;
+        default:
+            return GDC_ATTR_BLACK;
```
Comment (disean) [inline boot/freeldr/freeldr/arch/i386/pc98/pc98video.c:275]:
```suggestion
        case COLOR_BLACK:
            return GDC_ATTR_BLACK;
        case COLOR_BLUE:
            return GDC_ATTR_BLUE;
        case COLOR_GREEN:
        case COLOR_LIGHTGREEN:
            return GDC_ATTR_GREEN;
        case COLOR_CYAN:
        case COLOR_GRAY:
        case COLOR_DARKGRAY:
        case COLOR_WHITE:
            return GDC_ATTR_WHITE;
        case COLOR_RED:
        case COLOR_LIGHTRED:
            return GDC_ATTR_RED;
        case COLOR_MAGENTA:
        case COLOR_LIGHTMAGENTA:
            return GDC_ATTR_PURPLE;
        case COLOR_BROWN:
        case COLOR_YELLOW:
            return GDC_ATTR_YELLOW;
        case COLOR_LIGHTBLUE:
        case COLOR_LIGHTCYAN:
            return GDC_ATTR_LIGHTBLUE;
        default:
            return GDC_ATTR_BLACK;
```

File: boot/freeldr/freeldr/arch/i386/pc98/pc98video.c [hunk 3]
```diff
@@ -242,9 +322,46 @@ Pc98VideoPutChar(int Ch, UCHAR Attr, unsigned X, unsigned Y)
     UCHAR I = (Attr & 0x80) ? 0xFF : 0;
     ULONG VramOffset = X + (Y * CHAR_HEIGHT) * BYTES_PER_SCANLINE;
     PUCHAR FontPtr = BitmapFont8x16 + Ch * 16;
+    BOOLEAN CGFont = UseCGFont && (Ch != 0xB1 && Ch != 0xB2);
```
Comment (disean) [inline boot/freeldr/freeldr/arch/i386/pc98/pc98video.c:325]:
```suggestion
    BOOLEAN CGFont = UseCGFont && (Ch != LIGHT_FILL && Ch != MEDIUM_FILL && Ch != DARK_FILL);
```

File: boot/freeldr/freeldr/arch/i386/pc98/pc98video.c [hunk 4]
```diff
@@ -242,9 +322,46 @@ Pc98VideoPutChar(int Ch, UCHAR Attr, unsigned X, unsigned Y)
...
+        else
+        {
+            /* Set needed character code to obtain glyph from CG Window */
+            WRITE_PORT_UCHAR((PUCHAR)KCG_IO_o_CHARCODE_LOW, Ch);
+            WRITE_PORT_UCHAR((PUCHAR)KCG_IO_o_CHARCODE_HIGH, Ch >> 8);
+        }
+    }
+    else if (UseCGFont && CGAccelDraw)
+    {
+        /* Clear character at this place in Video RAM */
```
Comment (disean) [inline boot/freeldr/freeldr/arch/i386/pc98/pc98video.c:344]:
```suggestion
        /* Clear character at this place in Text RAM */
```

File: boot/freeldr/freeldr/arch/i386/pc98/pc98video.c [hunk 5]
```diff
@@ -336,4 +453,14 @@ Pc98VideoPrepareForReactOS(VOID)
     Int386(0x18, &Regs, &Regs);
 
     Pc98VideoHideShowTextCursor(FALSE);
+
+    if (UseCGFont && CGAccelDraw)
+    {
+        /* Clear the screen, resetting to default attributes */
```
Comment (disean) [inline boot/freeldr/freeldr/arch/i386/pc98/pc98video.c:459]:
```suggestion
        /* Clear the text screen, resetting to default attributes */
```

**Review by disean — APPROVED** [review]
Some minor suggestions

**Review by HBelusca — APPROVED** [review]

**Pull request merged**